### PR TITLE
fix(ui/textfield): formik version type options

### LINKED ISSUE #133: [UI core library] TextFieldFormik doesn't accept types other than "text"

### PR DESCRIPTION
**Pull request by jozefhruska**
Closes #133

### CODE REVIEW

File: packages/ui/src/TextFieldFormik.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import React, { ReactElement } from 'react'
 import { useField } from 'formik'
 
-import { TextField, TextFieldExtraProps } from './TextField'
+import { TextField, TextFieldExtraProps, TextFieldTypes } from './TextField'
 import { FieldValidatorGeneric, getFieldError } from './utils/formik'
 import { ExtractKeysOfValueType } from './utils/types'
 
-export type TextFieldFormikProps<V extends object> = Exclude<TextFieldExtraProps<'text'>, 'type'> & {
+export type TextFieldFormikProps<V extends object> = TextFieldExtraProps<TextFieldTypes> & {
```
Comment (aigoncharov) [inline packages/ui/src/TextFieldFormik.tsx:8]: We should not allow `number` as the type. I think we should allow `'email' | 'search' | 'tel' | 'text' | 'url' | undefined` only.

BTW, we should pass the proper type on L22

Comment (jozefhruska) [reply packages/ui/src/TextFieldFormik.tsx:8]: @aigoncharov Wow, good catch! But why shouldn’t we allow `number` type?

Comment (aigoncharov) [reply packages/ui/src/TextFieldFormik.tsx:8]: @jozefhruska we have `NumberField` for it, right? And for `password` we have `PasswordField`.

Comment (jozefhruska) [reply packages/ui/src/TextFieldFormik.tsx:8]: Cool, now I understand 👌 Will make the change, thanks.

**Review by aigoncharov — APPROVED** [review]

**Pull request merged**